_search-plugins/search-pipelines/index.md

```diff
@@ -0,0 +1,387 @@
+---
+layout: default
+title: Search pipelines
+nav_order: 100
+has_children: true
+has_toc: false
+---
+
+# Search pipelines
+
+Search pipelines is an experimental feature. Therefore, we do not recommend the use of search pipelines in a production environment. For updates on the progress of search pipelines, or if you want to leave feedback that could help improve the feature, see the associated [GitHub issue](https://github.com/opensearch-project/OpenSearch/issues/6278).    
+{: .warning}
+
+To integrate result rerankers, query rewriters, and other components that operate on queries or results, you can use _search pipelines_. Search pipelines make it easier for you to process search queries and search results right in OpenSearch, as opposed to building plugins or creating additional logic elsewhere in your application. Thus, you can customize search results without adding complexity to your application. Search pipelines also provide a mechanism for modularizing your processors so you can add and reorder them easily. 
+
+## Terminology
+
+The following is a list of search pipeline terminology:
+
+* _Search request processor_: A component that takes a search request (the query and the metadata passed in the request), performs some operation with or on the search request, and returns a search request.
+* _Search response processor_: A component that takes a search response and search request (the query, results, and metadata passed in the request), performs some operation with or on the search response, and returns a search response.
+* _Processor_: Either a search request processor or a search response processor.
+* _Search pipeline_: An ordered list of processors that is integrated into OpenSearch. The pipeline intercepts a query, performs processing on the query, sends it to OpenSearch, intercepts the results, performs processing on the results, and returns them to the calling application, as shown in the following diagram. 
+
+![Search processor diagram]({{site.url}}{{site.baseurl}}/images/search-pipelines.png)
+
+Both request and response processing for the pipeline are performed on the coordinator node so there is no shard-level processing.
```

Comment:
Tip: If the meaning of "so" is "so that," it doesn't take a comma, but if it means "therefore," as it does here, it takes a comma.